app/jenkins-for-jira-ui/src/components/JenkinsSetup/JenkinsSetup.tsx

```diff
@@ -219,37 +241,67 @@ const JenkinsSetup = (): JSX.Element => {
 		fetchData();
 	}, [uuid, getServer]);
 
-	const handleCopyToClipboard = (copyRef: React.RefObject<HTMLDivElement>) => {
-		if (copyRef.current) {
-			const range = document.createRange();
-			range.selectNode(copyRef.current);
-
-			const selection = window.getSelection();
-			if (selection) {
-				selection.removeAllRanges();
-				selection.addRange(range);
+	const handleCopyToClipboard =
+		async (copyRef: React.RefObject<HTMLDivElement>, elementName?: string) => {
+			await analyticsClient.sendAnalytics(
+				AnalyticsEventTypes.UiEvent,
+				AnalyticsUiEventsEnum.CopiedToClipboardName,
+				{
+					source: AnalyticsScreenEventsEnum.JenkinsSetupScreenName,
+					buttonClicked: elementName || 'Copy button'
+				}
+			);
+
+			if (copyRef.current) {
```

Comment:
Is there is Forge reasons that we use `document.execCommand` over `navigator.clipboard.writeText` :shakes-fist:

Reply:
Not a Forge thing. Ran into more issues testing in diff browsers with the latter.